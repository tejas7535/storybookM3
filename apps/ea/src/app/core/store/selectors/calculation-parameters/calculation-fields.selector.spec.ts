import {
  CALCULATION_PARAMETERS_STATE_MOCK,
  PRODUCT_SELECTION_STATE_MOCK,
} from '@ea/testing/mocks';

import { getCalculationFieldsConfig } from './calculation-fields.selector';

describe('Calculation Fields Selector', () => {
  const mockState = {
    productSelection: {
      ...PRODUCT_SELECTION_STATE_MOCK,
    },
    calculationParameters: {
      ...CALCULATION_PARAMETERS_STATE_MOCK,
    },
  };

  describe('getCalculationFieldsConfig', () => {
    it('should return the fields config', () => {
      expect(
        getCalculationFieldsConfig({
          ...mockState,
          productSelection: {
            ...mockState.productSelection,
            co2DownstreamAvailable: true,
          },
        })
      ).toMatchSnapshot();
    });

    it('should return the fields config without downstream fields', () => {
      expect(
        getCalculationFieldsConfig({
          ...mockState,
          productSelection: {
            ...mockState.productSelection,
            co2DownstreamAvailable: false,
          },
        })
      ).toMatchSnapshot();
    });
  });
});
