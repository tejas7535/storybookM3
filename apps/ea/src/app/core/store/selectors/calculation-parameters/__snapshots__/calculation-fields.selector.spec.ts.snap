// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculation Fields Selector getCalculationFieldsConfig should return the fields config 1`] = `
Object {
  "preset": Array [
    "energySource",
    "time",
    "operatingTemperature",
    "contamination",
  ],
  "required": Array [
    "load",
    "rotatingCondition",
    "lubrication",
  ],
}
`;

exports[`Calculation Fields Selector getCalculationFieldsConfig should return the fields config without downstream fields 1`] = `
Object {
  "preset": Array [
    "operatingTemperature",
    "contamination",
  ],
  "required": Array [
    "load",
    "rotatingCondition",
    "lubrication",
  ],
}
`;
